### PR TITLE
Add App Lock settings screen

### PR DESCRIPTION
**Pull request by pixlwave**
Adds the App Lock settings screen. This is the screen only, more screens are needed before the flow will work. This branch is based upon #1912.

Screenshot 2023-10-18 at 10 05 14 am

### CODE REVIEW

File: ElementX/Sources/Screens/AppLock/AppLockSettingsScreen/AppLockSettingsScreenViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        state.bindings.alertInfo = .init(id: .confirmRemovePINCode,
+                                         title: UntranslatedL10n.screenAppLockSettingsRemovePinAlertTitle,
+                                         message: UntranslatedL10n.screenAppLockSettingsRemovePinAlertMessage,
+                                         primaryButton: .init(title: L10n.actionYes) { self.completeRemovePIN() },
+                                         secondaryButton: .init(title: L10n.actionCancel, role: .cancel, action: nil))
+    }
+    
+    /// Removes the user's PIN code, disabling the App Lock feature.
+    private func completeRemovePIN() {
+        appLockService.disable()
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/AppLock/AppLockSettingsScreen/AppLockSettingsScreenViewModel.swift:66]: Shouldn't we ask the user to input the pin again here?

**Review by stefanceriu — APPROVED** [review]
lgtm 👍

File: ElementX/Sources/Screens/AppLock/AppLockSettingsScreen/AppLockSettingsScreenViewModel.swift (hunk 1 above)
Comment (pixlwave) [reply ElementX/Sources/Screens/AppLock/AppLockSettingsScreen/AppLockSettingsScreenViewModel.swift:66]: So in the flow, this whole screen will be protected by the user first re-entering their PIN. I need to build that screen before I can insert it into the flow though 🙈

**Comment by SpyHoodle** [comment]
Shouldn't this page be titled "PIN settings" or something, instead of just Settings, as you'd be going back to Settings?
image

**Pull request merged**